.github/workflows/golangci-lint.yml

```
@@ -0,0 +1,42 @@
+name: golangci-lint
+
+on:
+  pull_request:
```

Sugiro que adicione a seguinte alteração no codigo para somente executar o workflow caso o pr esteja direcionado para a main, pois sem essa especificação todo pr aberto poderá rodar esse workflow

```suggestion
  pull_request:
     branches: 
       - "main"
```

---

Feito